diff -U 3 -H -d -r -N -- Ultracopier/CopyEngine.cpp Rsync/CopyEngine.cpp
--- Ultracopier/CopyEngine.cpp	2013-03-27 01:39:20.000000000 +0100
+++ Rsync/CopyEngine.cpp	2013-03-28 16:47:29.023634963 +0100
@@ -222,6 +222,7 @@
     setSequentialBuffer(sequentialBuffer);
     setParallelBuffer(parallelBuffer);
     setAutoStart(autoStart);
+    setRsync(rsync);
     setCheckDestinationFolderExists(checkDestinationFolderExists);
     set_doChecksum(doChecksum);
     set_checksumIgnoreIfImpossible(checksumIgnoreIfImpossible);
@@ -358,6 +359,7 @@
         connect(ui->deletePartiallyTransferredFiles,    &QCheckBox::toggled,		this,&CopyEngine::setDeletePartiallyTransferredFiles);
         connect(ui->followTheStrictOrder,               &QCheckBox::toggled,        this,&CopyEngine::setFollowTheStrictOrder);
         connect(ui->checkBoxDestinationFolderExists,	&QCheckBox::toggled,        this,&CopyEngine::setCheckDestinationFolderExists);
+        connect(ui->rsync,                              &QCheckBox::toggled,        this,&CopyEngine::setRsync);
         connect(ui->renameTheOriginalDestination,       &QCheckBox::toggled,        this,&CopyEngine::setRenameTheOriginalDestination);
         connect(filters,                                &Filters::haveNewFilters,   this,&CopyEngine::sendNewFilters);
         connect(ui->filters,                            &QPushButton::clicked,      this,&CopyEngine::showFilterDialog);
@@ -670,16 +672,27 @@
 /** \brief give the forced mode, to export/import transfer list */
 void CopyEngine::forceMode(const Ultracopier::CopyMode &mode)
 {
+    if(mode==Ultracopier::Move)
+    {
+        listThread->setRsync(false);
+        rsync=false;
+    }
+    if(uiIsInstalled)
+        ui->rsync->setEnabled(mode==Ultracopier::Copy);
     if(forcedMode)
     {
         ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,QString("Mode forced previously"));
         QMessageBox::critical(NULL,facilityEngine->translateText("Internal error"),tr("The mode have been forced previously, it's internal error, please report it"));
         return;
     }
+    if(mode==Ultracopier::Move)
+        rsync=false;
     if(mode==Ultracopier::Copy)
         ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,QString("Force mode to copy"));
     else
         ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,QString("Force mode to move"));
+    if(uiIsInstalled)
+        ui->rsync->setEnabled(mode==Ultracopier::Copy);
     this->mode=mode;
     forcedMode=true;
     emit signal_forceMode(mode);
@@ -924,6 +937,20 @@
     listThread->setAutoStart(autoStart);
 }
 
+/// \brief set rsync
+void CopyEngine::setRsync(const bool rsync)
+{
+    ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,"set rsync: "+QString::number(rsync));
+    this->rsync=rsync;
+    if(uiIsInstalled)
+    {
+        ui->rsync->setChecked(rsync);
+        ui->rsync->setEnabled(forcedMode && mode==Ultracopier::Copy);
+        ui->label_rsync->setEnabled(forcedMode && mode==Ultracopier::Copy);
+    }
+    listThread->setRsync(rsync);
+}
+
 //set check destination folder
 void CopyEngine::setCheckDestinationFolderExists(const bool checkDestinationFolderExists)
 {
diff -U 3 -H -d -r -N -- Ultracopier/CopyEngine.h Rsync/CopyEngine.h
--- Ultracopier/CopyEngine.h	2013-03-26 13:57:31.000000000 +0100
+++ Rsync/CopyEngine.h	2013-03-27 09:48:38.828200695 +0100
@@ -67,6 +67,7 @@
     bool                    renameTheOriginalDestination;
     bool                    moveTheWholeFolder;
     bool                    autoStart;
+    bool                    rsync;
     bool                    checkDestinationFolderExists;
     FileExistsAction		alwaysDoThisActionForFileExists;
     FileErrorAction			alwaysDoThisActionForFileError;
@@ -229,6 +230,7 @@
     void set_osBufferLimit(unsigned int osBufferLimit);
     void set_setFilters(QStringList includeStrings,QStringList includeOptions,QStringList excludeStrings,QStringList excludeOptions);
     void setRenamingRules(QString firstRenamingRule,QString otherRenamingRule);
+    void setRsync(const bool rsync);
     void setCheckDiskSpace(const bool &checkDiskSpace);
     void setDefaultDestinationFolder(const QString &defaultDestinationFolder);
     void defaultDestinationFolderBrowse();
diff -U 3 -H -d -r -N -- Ultracopier/CopyEngine.pro.user Rsync/CopyEngine.pro.user
--- Ultracopier/CopyEngine.pro.user	1970-01-01 01:00:00.000000000 +0100
+++ Rsync/CopyEngine.pro.user	2013-03-28 17:05:30.417590694 +0100
@@ -0,0 +1,208 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE QtCreatorProject>
+<!-- Written by Qt Creator 2.6.2, 2013-03-28T17:05:30. -->
+<qtcreator>
+ <data>
+  <variable>ProjectExplorer.Project.ActiveTarget</variable>
+  <value type="int">0</value>
+ </data>
+ <data>
+  <variable>ProjectExplorer.Project.EditorSettings</variable>
+  <valuemap type="QVariantMap">
+   <value type="bool" key="EditorConfiguration.AutoIndent">true</value>
+   <value type="bool" key="EditorConfiguration.AutoSpacesForTabs">false</value>
+   <value type="bool" key="EditorConfiguration.CamelCaseNavigation">true</value>
+   <valuemap type="QVariantMap" key="EditorConfiguration.CodeStyle.0">
+    <value type="QString" key="language">Cpp</value>
+    <valuemap type="QVariantMap" key="value">
+     <value type="QString" key="CurrentPreferences">CppGlobal</value>
+    </valuemap>
+   </valuemap>
+   <valuemap type="QVariantMap" key="EditorConfiguration.CodeStyle.1">
+    <value type="QString" key="language">QmlJS</value>
+    <valuemap type="QVariantMap" key="value">
+     <value type="QString" key="CurrentPreferences">QmlJSGlobal</value>
+    </valuemap>
+   </valuemap>
+   <value type="int" key="EditorConfiguration.CodeStyle.Count">2</value>
+   <value type="QByteArray" key="EditorConfiguration.Codec">UTF-8</value>
+   <value type="bool" key="EditorConfiguration.ConstrainTooltips">false</value>
+   <value type="int" key="EditorConfiguration.IndentSize">4</value>
+   <value type="bool" key="EditorConfiguration.KeyboardTooltips">false</value>
+   <value type="bool" key="EditorConfiguration.MouseNavigation">true</value>
+   <value type="int" key="EditorConfiguration.PaddingMode">1</value>
+   <value type="bool" key="EditorConfiguration.ScrollWheelZooming">true</value>
+   <value type="int" key="EditorConfiguration.SmartBackspaceBehavior">0</value>
+   <value type="bool" key="EditorConfiguration.SpacesForTabs">true</value>
+   <value type="int" key="EditorConfiguration.TabKeyBehavior">0</value>
+   <value type="int" key="EditorConfiguration.TabSize">8</value>
+   <value type="bool" key="EditorConfiguration.UseGlobal">true</value>
+   <value type="int" key="EditorConfiguration.Utf8BomBehavior">1</value>
+   <value type="bool" key="EditorConfiguration.addFinalNewLine">true</value>
+   <value type="bool" key="EditorConfiguration.cleanIndentation">true</value>
+   <value type="bool" key="EditorConfiguration.cleanWhitespace">true</value>
+   <value type="bool" key="EditorConfiguration.inEntireDocument">true</value>
+  </valuemap>
+ </data>
+ <data>
+  <variable>ProjectExplorer.Project.PluginSettings</variable>
+  <valuemap type="QVariantMap"/>
+ </data>
+ <data>
+  <variable>ProjectExplorer.Project.Target.0</variable>
+  <valuemap type="QVariantMap">
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Desktop-Qt 5.01 (qt5-linux)</value>
+   <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName">Desktop-Qt 5.01 (qt5-linux)</value>
+   <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">{55d7d977-66a3-4d72-9331-2ae779c6252a}</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveBuildConfiguration">0</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveDeployConfiguration">0</value>
+   <value type="int" key="ProjectExplorer.Target.ActiveRunConfiguration">0</value>
+   <valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.0">
+    <valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.0">
+     <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.0">
+      <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">qmake</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+      <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">QtProjectManager.QMakeBuildStep</value>
+      <value type="bool" key="QtProjectManager.QMakeBuildStep.LinkQmlDebuggingLibrary">false</value>
+      <value type="bool" key="QtProjectManager.QMakeBuildStep.LinkQmlDebuggingLibraryAuto">true</value>
+      <value type="QString" key="QtProjectManager.QMakeBuildStep.QMakeArguments"></value>
+      <value type="bool" key="QtProjectManager.QMakeBuildStep.QMakeForced">false</value>
+     </valuemap>
+     <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.1">
+      <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Make</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+      <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">Qt4ProjectManager.MakeStep</value>
+      <value type="bool" key="Qt4ProjectManager.MakeStep.Clean">false</value>
+      <value type="QString" key="Qt4ProjectManager.MakeStep.MakeArguments"></value>
+      <value type="QString" key="Qt4ProjectManager.MakeStep.MakeCommand"></value>
+     </valuemap>
+     <value type="int" key="ProjectExplorer.BuildStepList.StepsCount">2</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Build</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+     <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">ProjectExplorer.BuildSteps.Build</value>
+    </valuemap>
+    <valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.1">
+     <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.0">
+      <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Make</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+      <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">Qt4ProjectManager.MakeStep</value>
+      <value type="bool" key="Qt4ProjectManager.MakeStep.Clean">true</value>
+      <value type="QString" key="Qt4ProjectManager.MakeStep.MakeArguments">clean</value>
+      <value type="QString" key="Qt4ProjectManager.MakeStep.MakeCommand"></value>
+     </valuemap>
+     <value type="int" key="ProjectExplorer.BuildStepList.StepsCount">1</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Clean</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+     <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">ProjectExplorer.BuildSteps.Clean</value>
+    </valuemap>
+    <value type="int" key="ProjectExplorer.BuildConfiguration.BuildStepListCount">2</value>
+    <value type="bool" key="ProjectExplorer.BuildConfiguration.ClearSystemEnvironment">false</value>
+    <valuelist type="QVariantList" key="ProjectExplorer.BuildConfiguration.UserEnvironmentChanges"/>
+    <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Debug</value>
+    <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+    <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">Qt4ProjectManager.Qt4BuildConfiguration</value>
+    <value type="int" key="Qt4ProjectManager.Qt4BuildConfiguration.BuildConfiguration">2</value>
+    <value type="QString" key="Qt4ProjectManager.Qt4BuildConfiguration.BuildDirectory">/home/user/Desktop/ultracopier/sources/plugins-alternative/CopyEngine/CopyEngine-build-Desktop_Qt_5_01_qt5_linux-Debug</value>
+    <value type="bool" key="Qt4ProjectManager.Qt4BuildConfiguration.UseShadowBuild">true</value>
+   </valuemap>
+   <valuemap type="QVariantMap" key="ProjectExplorer.Target.BuildConfiguration.1">
+    <valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.0">
+     <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.0">
+      <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">qmake</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+      <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">QtProjectManager.QMakeBuildStep</value>
+      <value type="bool" key="QtProjectManager.QMakeBuildStep.LinkQmlDebuggingLibrary">false</value>
+      <value type="bool" key="QtProjectManager.QMakeBuildStep.LinkQmlDebuggingLibraryAuto">true</value>
+      <value type="QString" key="QtProjectManager.QMakeBuildStep.QMakeArguments"></value>
+      <value type="bool" key="QtProjectManager.QMakeBuildStep.QMakeForced">false</value>
+     </valuemap>
+     <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.1">
+      <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Make</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+      <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">Qt4ProjectManager.MakeStep</value>
+      <value type="bool" key="Qt4ProjectManager.MakeStep.Clean">false</value>
+      <value type="QString" key="Qt4ProjectManager.MakeStep.MakeArguments"></value>
+      <value type="QString" key="Qt4ProjectManager.MakeStep.MakeCommand"></value>
+     </valuemap>
+     <value type="int" key="ProjectExplorer.BuildStepList.StepsCount">2</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Build</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+     <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">ProjectExplorer.BuildSteps.Build</value>
+    </valuemap>
+    <valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.1">
+     <valuemap type="QVariantMap" key="ProjectExplorer.BuildStepList.Step.0">
+      <value type="bool" key="ProjectExplorer.BuildStep.Enabled">true</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Make</value>
+      <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+      <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">Qt4ProjectManager.MakeStep</value>
+      <value type="bool" key="Qt4ProjectManager.MakeStep.Clean">true</value>
+      <value type="QString" key="Qt4ProjectManager.MakeStep.MakeArguments">clean</value>
+      <value type="QString" key="Qt4ProjectManager.MakeStep.MakeCommand"></value>
+     </valuemap>
+     <value type="int" key="ProjectExplorer.BuildStepList.StepsCount">1</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Clean</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+     <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">ProjectExplorer.BuildSteps.Clean</value>
+    </valuemap>
+    <value type="int" key="ProjectExplorer.BuildConfiguration.BuildStepListCount">2</value>
+    <value type="bool" key="ProjectExplorer.BuildConfiguration.ClearSystemEnvironment">false</value>
+    <valuelist type="QVariantList" key="ProjectExplorer.BuildConfiguration.UserEnvironmentChanges"/>
+    <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Release</value>
+    <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+    <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">Qt4ProjectManager.Qt4BuildConfiguration</value>
+    <value type="int" key="Qt4ProjectManager.Qt4BuildConfiguration.BuildConfiguration">0</value>
+    <value type="QString" key="Qt4ProjectManager.Qt4BuildConfiguration.BuildDirectory">/home/user/Desktop/ultracopier/sources/plugins-alternative/CopyEngine/CopyEngine-build-Desktop_Qt_5_01_qt5_linux-Release</value>
+    <value type="bool" key="Qt4ProjectManager.Qt4BuildConfiguration.UseShadowBuild">true</value>
+   </valuemap>
+   <value type="int" key="ProjectExplorer.Target.BuildConfigurationCount">2</value>
+   <valuemap type="QVariantMap" key="ProjectExplorer.Target.DeployConfiguration.0">
+    <valuemap type="QVariantMap" key="ProjectExplorer.BuildConfiguration.BuildStepList.0">
+     <value type="int" key="ProjectExplorer.BuildStepList.StepsCount">0</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Deploy</value>
+     <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+     <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">ProjectExplorer.BuildSteps.Deploy</value>
+    </valuemap>
+    <value type="int" key="ProjectExplorer.BuildConfiguration.BuildStepListCount">1</value>
+    <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Deploy locally</value>
+    <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+    <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">ProjectExplorer.DefaultDeployConfiguration</value>
+   </valuemap>
+   <value type="int" key="ProjectExplorer.Target.DeployConfigurationCount">1</value>
+   <valuemap type="QVariantMap" key="ProjectExplorer.Target.RunConfiguration.0">
+    <value type="bool" key="Analyzer.Project.UseGlobal">true</value>
+    <value type="QString" key="ProjectExplorer.CustomExecutableRunConfiguration.Arguments"></value>
+    <value type="int" key="ProjectExplorer.CustomExecutableRunConfiguration.BaseEnvironmentBase">2</value>
+    <value type="QString" key="ProjectExplorer.CustomExecutableRunConfiguration.Executable"></value>
+    <value type="bool" key="ProjectExplorer.CustomExecutableRunConfiguration.UseTerminal">false</value>
+    <valuelist type="QVariantList" key="ProjectExplorer.CustomExecutableRunConfiguration.UserEnvironmentChanges"/>
+    <value type="QString" key="ProjectExplorer.CustomExecutableRunConfiguration.WorkingDirectory">%{buildDir}</value>
+    <value type="QString" key="ProjectExplorer.ProjectConfiguration.DefaultDisplayName">Custom Executable</value>
+    <value type="QString" key="ProjectExplorer.ProjectConfiguration.DisplayName"></value>
+    <value type="QByteArray" key="ProjectExplorer.ProjectConfiguration.Id">ProjectExplorer.CustomExecutableRunConfiguration</value>
+    <value type="uint" key="RunConfiguration.QmlDebugServerPort">3768</value>
+    <value type="bool" key="RunConfiguration.UseCppDebugger">true</value>
+    <value type="bool" key="RunConfiguration.UseMultiProcess">false</value>
+    <value type="bool" key="RunConfiguration.UseQmlDebugger">false</value>
+    <value type="bool" key="RunConfiguration.UseQmlDebuggerAuto">true</value>
+   </valuemap>
+   <value type="int" key="ProjectExplorer.Target.RunConfigurationCount">1</value>
+  </valuemap>
+ </data>
+ <data>
+  <variable>ProjectExplorer.Project.TargetCount</variable>
+  <value type="int">1</value>
+ </data>
+ <data>
+  <variable>ProjectExplorer.Project.Updater.EnvironmentId</variable>
+  <value type="QString">{74ab603f-f657-4135-92cf-c93af71b2f91}</value>
+ </data>
+ <data>
+  <variable>ProjectExplorer.Project.Updater.FileVersion</variable>
+  <value type="int">12</value>
+ </data>
+</qtcreator>
diff -U 3 -H -d -r -N -- Ultracopier/DiskSpace.ui Rsync/DiskSpace.ui
--- Ultracopier/DiskSpace.ui	2013-03-27 12:00:44.000000000 +0100
+++ Rsync/DiskSpace.ui	2013-03-27 12:03:01.627870627 +0100
@@ -15,7 +15,7 @@
   </property>
   <property name="windowIcon">
    <iconset resource="copyEngineResources.qrc">
-    <normaloff>:/CopyEngine/Ultracopier/resources/remove.png</normaloff>:/CopyEngine/Ultracopier/resources/remove.png</iconset>
+    <normaloff>:/CopyEngine/Ultracopier/Rsync/remove.png</normaloff>:/CopyEngine/Ultracopier/Rsync/remove.png</iconset>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
diff -U 3 -H -d -r -N -- Ultracopier/Factory.cpp Rsync/Factory.cpp
--- Ultracopier/Factory.cpp	2013-03-26 13:56:20.000000000 +0100
+++ Rsync/Factory.cpp	2013-03-27 09:52:11.069192007 +0100
@@ -56,6 +56,7 @@
     connect(ui->osBuffer,                   &QCheckBox::toggled,                this,&CopyEngineFactory::osBuffer_toggled);
     connect(ui->osBufferLimited,            &QCheckBox::toggled,                this,&CopyEngineFactory::osBufferLimited_toggled);
     connect(ui->osBufferLimit,              &QSpinBox::editingFinished,         this,&CopyEngineFactory::osBufferLimit_editingFinished);
+    connect(ui->rsync,                      &QCheckBox::toggled,                this,&CopyEngineFactory::setRsync);
     connect(ui->inodeThreads,               &QSpinBox::editingFinished,         this,&CopyEngineFactory::on_inodeThreads_editingFinished);
     connect(ui->osBufferLimited,            &QAbstractButton::toggled,          this,&CopyEngineFactory::updateBufferCheckbox);
     connect(ui->osBuffer,                   &QAbstractButton::toggled,          this,&CopyEngineFactory::updateBufferCheckbox);
@@ -107,6 +108,7 @@
     realObject->setKeepDate(ui->keepDate->isChecked());
     realObject->setBlockSize(ui->blockSize->value());
     realObject->setAutoStart(ui->autoStart->isChecked());
+    realObject->setRsync(ui->rsync->isChecked());
     realObject->setFolderCollision(ui->comboBoxFolderCollision->currentIndex());
     realObject->setFolderError(ui->comboBoxFolderError->currentIndex());
     realObject->setFileCollision(ui->comboBoxFileCollision->currentIndex());
@@ -163,6 +165,7 @@
         KeysList.append(qMakePair(QString("parallelBuffer"),QVariant(ULTRACOPIER_PLUGIN_DEFAULT_BLOCK_SIZE*ULTRACOPIER_PLUGIN_DEFAULT_PARALLEL_NUMBER_OF_BLOCK)));
         KeysList.append(qMakePair(QString("parallelizeIfSmallerThan"),QVariant(1)));
         KeysList.append(qMakePair(QString("autoStart"),QVariant(true)));
+        KeysList.append(qMakePair(QString("rsync"),QVariant(true)));
         KeysList.append(qMakePair(QString("folderError"),QVariant(0)));
         KeysList.append(qMakePair(QString("folderCollision"),QVariant(0)));
         KeysList.append(qMakePair(QString("fileError"),QVariant(0)));
@@ -201,6 +204,7 @@
         ui->keepDate->setChecked(options->getOptionValue("keepDate").toBool());
         ui->blockSize->setValue(options->getOptionValue("blockSize").toUInt());//keep before sequentialBuffer and parallelBuffer
         ui->autoStart->setChecked(options->getOptionValue("autoStart").toBool());
+        ui->rsync->setChecked(options->getOptionValue("rsync").toBool());
         ui->comboBoxFolderError->setCurrentIndex(options->getOptionValue("folderError").toUInt());
         ui->comboBoxFolderCollision->setCurrentIndex(options->getOptionValue("folderCollision").toUInt());
         ui->comboBoxFileError->setCurrentIndex(options->getOptionValue("fileError").toUInt());
@@ -639,3 +643,10 @@
     if(optionsEngine!=NULL)
         optionsEngine->setOptionValue("inodeThreads",ui->inodeThreads->value());
 }
+
+void CopyEngineFactory::setRsync(bool rsync)
+{
+    ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,"the value have changed");
+    if(optionsEngine!=NULL)
+        optionsEngine->setOptionValue("rsync",rsync);
+}
diff -U 3 -H -d -r -N -- Ultracopier/Factory.h Rsync/Factory.h
--- Ultracopier/Factory.h	2013-03-26 13:54:25.000000000 +0100
+++ Rsync/Factory.h	2013-03-27 09:52:44.891190622 +0100
@@ -77,6 +77,7 @@
     void setSequentialBuffer(int sequentialBuffer);
     void setParallelizeIfSmallerThan(int parallelizeIfSmallerThan);
     void setAutoStart(bool autoStart);
+    void setRsync(bool rsync);
     void setFolderCollision(int index);
     void setFolderError(int index);
     void setTransferAlgorithm(int index);
diff -U 3 -H -d -r -N -- Ultracopier/FilterRules.ui Rsync/FilterRules.ui
--- Ultracopier/FilterRules.ui	2013-03-27 12:00:07.000000000 +0100
+++ Rsync/FilterRules.ui	2013-03-27 12:02:39.664871526 +0100
@@ -18,7 +18,7 @@
   </property>
   <property name="windowIcon">
    <iconset resource="resources.qrc">
-    <normaloff>:/CopyEngine/Ultracopier/resources/filter.png</normaloff>:/CopyEngine/Ultracopier/resources/filter.png</iconset>
+    <normaloff>:/CopyEngine/Ultracopier/Rsync/filter.png</normaloff>:/CopyEngine/Ultracopier/Rsync/filter.png</iconset>
   </property>
   <layout class="QGridLayout" name="gridLayout_2">
    <property name="margin">
diff -U 3 -H -d -r -N -- Ultracopier/Filters.ui Rsync/Filters.ui
--- Ultracopier/Filters.ui	2013-03-27 11:59:50.000000000 +0100
+++ Rsync/Filters.ui	2013-03-27 12:02:27.298872032 +0100
@@ -15,7 +15,7 @@
   </property>
   <property name="windowIcon">
    <iconset resource="resources.qrc">
-    <normaloff>:/CopyEngine/Ultracopier/resources/filter.png</normaloff>:/CopyEngine/Ultracopier/resources/filter.png</iconset>
+    <normaloff>:/CopyEngine/Ultracopier/Rsync/filter.png</normaloff>:/CopyEngine/Ultracopier/Rsync/filter.png</iconset>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
@@ -49,7 +49,7 @@
          <widget class="QToolButton" name="add_exclusion">
           <property name="icon">
            <iconset resource="resources.qrc">
-            <normaloff>:/CopyEngine/Ultracopier/resources/add.png</normaloff>:/CopyEngine/Ultracopier/resources/add.png</iconset>
+            <normaloff>:/CopyEngine/Ultracopier/Rsync/add.png</normaloff>:/CopyEngine/Ultracopier/Rsync/add.png</iconset>
           </property>
          </widget>
         </item>
@@ -57,7 +57,7 @@
          <widget class="QToolButton" name="edit_exclusion">
           <property name="icon">
            <iconset resource="resources.qrc">
-            <normaloff>:/CopyEngine/Ultracopier/resources/edit.png</normaloff>:/CopyEngine/Ultracopier/resources/edit.png</iconset>
+            <normaloff>:/CopyEngine/Ultracopier/Rsync/edit.png</normaloff>:/CopyEngine/Ultracopier/Rsync/edit.png</iconset>
           </property>
          </widget>
         </item>
@@ -65,7 +65,7 @@
          <widget class="QToolButton" name="remove_exclusion">
           <property name="icon">
            <iconset resource="resources.qrc">
-            <normaloff>:/CopyEngine/Ultracopier/resources/remove.png</normaloff>:/CopyEngine/Ultracopier/resources/remove.png</iconset>
+            <normaloff>:/CopyEngine/Ultracopier/Rsync/remove.png</normaloff>:/CopyEngine/Ultracopier/Rsync/remove.png</iconset>
           </property>
          </widget>
         </item>
@@ -121,7 +121,7 @@
            <widget class="QToolButton" name="add_inclusion">
             <property name="icon">
              <iconset resource="resources.qrc">
-              <normaloff>:/CopyEngine/Ultracopier/resources/add.png</normaloff>:/CopyEngine/Ultracopier/resources/add.png</iconset>
+              <normaloff>:/CopyEngine/Ultracopier/Rsync/add.png</normaloff>:/CopyEngine/Ultracopier/Rsync/add.png</iconset>
             </property>
            </widget>
           </item>
@@ -129,7 +129,7 @@
            <widget class="QToolButton" name="edit_inclusion">
             <property name="icon">
              <iconset resource="resources.qrc">
-              <normaloff>:/CopyEngine/Ultracopier/resources/edit.png</normaloff>:/CopyEngine/Ultracopier/resources/edit.png</iconset>
+              <normaloff>:/CopyEngine/Ultracopier/Rsync/edit.png</normaloff>:/CopyEngine/Ultracopier/Rsync/edit.png</iconset>
             </property>
            </widget>
           </item>
@@ -137,7 +137,7 @@
            <widget class="QToolButton" name="remove_inclusion">
             <property name="icon">
              <iconset resource="resources.qrc">
-              <normaloff>:/CopyEngine/Ultracopier/resources/remove.png</normaloff>:/CopyEngine/Ultracopier/resources/remove.png</iconset>
+              <normaloff>:/CopyEngine/Ultracopier/Rsync/remove.png</normaloff>:/CopyEngine/Ultracopier/Rsync/remove.png</iconset>
             </property>
            </widget>
           </item>
diff -U 3 -H -d -r -N -- Ultracopier/ListThread.cpp Rsync/ListThread.cpp
--- Ultracopier/ListThread.cpp	2013-03-28 16:52:18.000000000 +0100
+++ Rsync/ListThread.cpp	2013-03-28 16:56:41.002612366 +0100
@@ -22,6 +22,7 @@
     inodeThreads                    = 1;
     renameTheOriginalDestination    = false;
     doRightTransfer                 = false;
+    rsync                           = false;
     keepDate                        = false;
     checkDiskSpace                  = true;
     blockSize                       = ULTRACOPIER_PLUGIN_DEFAULT_BLOCK_SIZE*1024;
@@ -262,6 +263,22 @@
     this->autoStart=autoStart;
 }
 
+/// \brief set rsync
+void ListThread::setRsync(const bool rsync)
+{
+    ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,"set rsync: "+QString::number(rsync));
+    this->rsync=rsync;
+    int index=0;
+    int loop_sub_size_transfer_thread_search=transferThreadList.size();
+    while(index<loop_sub_size_transfer_thread_search)
+    {
+        transferThreadList.at(index)->setRsync(rsync);
+        index++;
+    }
+    for(int i=0;i<scanFileOrFolderThreadsPool.size();i++)
+        scanFileOrFolderThreadsPool.at(i)->setRsync(rsync);
+}
+
 //set check destination folder
 void ListThread::setCheckDestinationFolderExists(const bool checkDestinationFolderExists)
 {
@@ -345,6 +362,7 @@
     connect(scanFileOrFolderThreadsPool.last(),&ScanFileOrFolder::addToMovePath,					this,&ListThread::addToMovePath,			Qt::QueuedConnection);
     connect(scanFileOrFolderThreadsPool.last(),&ScanFileOrFolder::addToRealMove,					this,&ListThread::addToRealMove,			Qt::QueuedConnection);
     connect(scanFileOrFolderThreadsPool.last(),&ScanFileOrFolder::addToMkPath,                      this,&ListThread::addToMkPath,			Qt::QueuedConnection);
+    connect(scanFileOrFolderThreadsPool.last(),&ScanFileOrFolder::addToRmForRsync,                  this,&ListThread::addToRmForRsync,			Qt::QueuedConnection);
 
     connect(scanFileOrFolderThreadsPool.last(),&ScanFileOrFolder::errorOnFolder,					this,&ListThread::errorOnFolder,		Qt::QueuedConnection);
     connect(scanFileOrFolderThreadsPool.last(),&ScanFileOrFolder::folderAlreadyExists,				this,&ListThread::folderAlreadyExists,		Qt::QueuedConnection);
@@ -352,6 +370,7 @@
     scanFileOrFolderThreadsPool.last()->setFilters(include,exclude);
     scanFileOrFolderThreadsPool.last()->setCheckDestinationFolderExists(checkDestinationFolderExists && alwaysDoThisActionForFolderExists!=FolderExists_Merge);
     scanFileOrFolderThreadsPool.last()->setMoveTheWholeFolder(moveTheWholeFolder);
+    scanFileOrFolderThreadsPool.last()->setRsync(rsync);
     if(scanFileOrFolderThreadsPool.size()==1)
         updateTheStatus();
     scanFileOrFolderThreadsPool.last()->setRenamingRules(firstRenamingRule,otherRenamingRule);
@@ -930,6 +949,18 @@
     actionToDoListInode << temp;
 }
 
+//rsync rm
+void ListThread::addToRmForRsync(const QFileInfo& destination)
+{
+    ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,"inode: "+destination.absoluteFilePath());
+    ActionToDoInode temp;
+    temp.type	= ActionType_RmSync;
+    temp.id		= generateIdNumber();
+    temp.destination= destination;
+    temp.isRunning	= false;
+    actionToDoListInode << temp;
+}
+
 //send action done
 void ListThread::sendActionDone()
 {
@@ -1286,6 +1317,8 @@
 /** \brief give the forced mode, to export/import transfer list */
 void ListThread::forceMode(const Ultracopier::CopyMode &mode)
 {
+    if(mode==Ultracopier::Move)
+        setRsync(false);
     if(mode==Ultracopier::Copy)
         ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,QString("Force mode to copy"));
     else
@@ -1774,10 +1807,11 @@
                 doNewActions_inode_manipulation();
                 return;
             }
-            if(actionToDoListInode.at(int_for_loop).type==ActionType_MovePath || actionToDoListInode.at(int_for_loop).type==ActionType_RealMove)
+            if(actionToDoListInode.at(int_for_loop).type==ActionType_MovePath || actionToDoListInode.at(int_for_loop).type==ActionType_RmSync || actionToDoListInode.at(int_for_loop).type==ActionType_RealMove)
             {
                 //to send to the log
-                emit mkPath(actionToDoListInode.at(int_for_loop).destination.absoluteFilePath());
+                if(actionToDoListInode.at(int_for_loop).type!=ActionType_RmSync)
+                    emit mkPath(actionToDoListInode.at(int_for_loop).destination.absoluteFilePath());
                 emit rmPath(actionToDoListInode.at(int_for_loop).source.absoluteFilePath());
                 ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,QString("stop mkpath: %1").arg(actionToDoListInode.at(int_for_loop).destination.absoluteFilePath()));
                 actionToDoListInode.removeAt(int_for_loop);
@@ -1956,6 +1990,7 @@
     last->set_osBufferLimited(osBufferLimited);
     last->set_osBufferLimit(osBufferLimit);
     last->setDeletePartiallyTransferredFiles(deletePartiallyTransferredFiles);
+    last->setRsync(rsync);
 
     #ifdef ULTRACOPIER_PLUGIN_DEBUG
     connect(last,&TransferThread::debugInformation,             this,&ListThread::debugInformation,             Qt::QueuedConnection);
diff -U 3 -H -d -r -N -- Ultracopier/ListThread.h Rsync/ListThread.h
--- Ultracopier/ListThread.h	2013-03-25 18:33:30.000000000 +0100
+++ Rsync/ListThread.h	2013-03-27 09:22:40.931264471 +0100
@@ -147,6 +147,8 @@
     void setBlockSize(const int blockSize);
     /// \brief set auto start
     void setAutoStart(const bool autoStart);
+    /// \brief set rsync
+    void setRsync(const bool rsync);
     /// \brief set check destination folder
     void setCheckDestinationFolderExists(const bool checkDestinationFolderExists);
     /// \brief set data local to the thread
@@ -204,6 +206,7 @@
     quint64				bytesToTransfer;
     quint64				bytesTransfered;
     bool				autoStart;
+    bool                rsync;
     bool				putInPause;
     QList<Ultracopier::ReturnActionOnCopyList>	actionDone;///< to action to send to the interface
     quint64				idIncrementNumber;///< to store the last id returned
@@ -329,6 +332,8 @@
     void addToMovePath(const QFileInfo& source,const QFileInfo& destination, const int& inodeToRemove);
     //add to real move
     void addToRealMove(const QFileInfo& source,const QFileInfo& destination);
+    //rsync rm
+    void addToRmForRsync(const QFileInfo& destination);
     //send the progression, after full reset of the interface (then all is empty)
     void syncTransferList_internal();
 signals:
diff -U 3 -H -d -r -N -- Ultracopier/ListThread_InodeAction.cpp Rsync/ListThread_InodeAction.cpp
--- Ultracopier/ListThread_InodeAction.cpp	2013-03-27 09:25:07.000000000 +0100
+++ Rsync/ListThread_InodeAction.cpp	2013-03-27 09:28:26.115250340 +0100
@@ -27,6 +27,14 @@
         if(numberOfInodeOperation>=inodeThreads)
             return;
     break;
+    case ActionType_RmSync:
+        ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,QString("launch rmsync, destination: %1").arg(currentActionToDoInode.destination.absoluteFilePath()));
+        mkPathQueue.addPath(currentActionToDoInode.destination.absoluteFilePath(),currentActionToDoInode.destination.absoluteFilePath(),currentActionToDoInode.type);
+        currentActionToDoInode.isRunning=true;
+        numberOfInodeOperation++;
+        if(numberOfInodeOperation>=inodeThreads)
+            return;
+    break;
     case ActionType_MovePath:
         //then empty (no file), can try remove it
         if(currentActionToDoInode.size==0)
diff -U 3 -H -d -r -N -- Ultracopier/MkPath.cpp Rsync/MkPath.cpp
--- Ultracopier/MkPath.cpp	2013-03-22 13:34:36.000000000 +0100
+++ Rsync/MkPath.cpp	2013-03-28 17:03:38.981595256 +0100
@@ -60,82 +60,110 @@
     if(waitAction || pathList.isEmpty())
         return;
     ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,QString("source: %1, destination: %2, move: %3").arg(pathList.first().source.absoluteFilePath()).arg(pathList.first().destination.absoluteFilePath()).arg(pathList.first().actionType));
-    doTheDateTransfer=false;
-    if(keepDate)
+    if(pathList.first().actionType==ActionType_RmSync)
     {
-        doTheDateTransfer=readFileDateTime(pathList.first().source);
-        if(!doTheDateTransfer)
+        if(pathList.first().destination.isFile())
+        {
+            QFile removedFile(pathList.first().destination.absoluteFilePath());
+            if(!removedFile.remove())
+            {
+                if(stopIt)
+                    return;
+                waitAction=true;
+                ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to remove the inode: "+pathList.first().destination.absoluteFilePath()+", error: "+removedFile.errorString());
+                emit errorOnFolder(pathList.first().destination,removedFile.errorString());
+                return;
+            }
+        }
+        else if(!rmpath(pathList.first().destination.absoluteFilePath()))
         {
             if(stopIt)
                 return;
             waitAction=true;
-            ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to get source folder time: "+pathList.first().destination.absoluteFilePath());
-            emit errorOnFolder(pathList.first().source,tr("Unable to get time"));
+            ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to remove the inode: "+pathList.first().destination.absoluteFilePath());
+            emit errorOnFolder(pathList.first().destination,tr("Unable to remove"));
             return;
         }
     }
-    if(pathList.first().actionType!=ActionType_RealMove)
+    else
     {
-        if(!dir.exists(pathList.first().destination.absoluteFilePath()))
-            if(!dir.mkpath(pathList.first().destination.absoluteFilePath()))
+        doTheDateTransfer=false;
+        if(keepDate)
+        {
+            doTheDateTransfer=readFileDateTime(pathList.first().source);
+            if(!doTheDateTransfer)
             {
-                if(!dir.exists(pathList.first().destination.absoluteFilePath()))
-                {
-                    if(stopIt)
-                        return;
-                    waitAction=true;
-                    ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to make the folder: "+pathList.first().destination.absoluteFilePath());
-                    emit errorOnFolder(pathList.first().destination,tr("Unable to create the folder"));
+                if(stopIt)
                     return;
-                }
+                waitAction=true;
+                ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to get source folder time: "+pathList.first().destination.absoluteFilePath());
+                emit errorOnFolder(pathList.first().source,tr("Unable to get time"));
+                return;
             }
-    }
-    else
-    {
-        if(!dir.rename(pathList.first().source.absoluteFilePath(),pathList.first().destination.absoluteFilePath()))
+        }
+        if(pathList.first().actionType!=ActionType_RealMove)
         {
-            if(stopIt)
-                return;
-            waitAction=true;
-            ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to make the folder: "+pathList.first().destination.absoluteFilePath());
-            emit errorOnFolder(pathList.first().destination,tr("Unable to create the folder"));
-            return;
+            if(!dir.exists(pathList.first().destination.absoluteFilePath()))
+                if(!dir.mkpath(pathList.first().destination.absoluteFilePath()))
+                {
+                    if(!dir.exists(pathList.first().destination.absoluteFilePath()))
+                    {
+                        if(stopIt)
+                            return;
+                        waitAction=true;
+                        ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to make the folder: "+pathList.first().destination.absoluteFilePath());
+                        emit errorOnFolder(pathList.first().destination,tr("Unable to create the folder"));
+                        return;
+                    }
+                }
         }
-    }
-    if(doTheDateTransfer)
-        if(!writeFileDateTime(pathList.first().destination))
+        else
         {
-            if(stopIt)
+            if(!dir.rename(pathList.first().source.absoluteFilePath(),pathList.first().destination.absoluteFilePath()))
+            {
+                if(stopIt)
+                    return;
+                waitAction=true;
+                ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to make the folder: "+pathList.first().destination.absoluteFilePath());
+                emit errorOnFolder(pathList.first().destination,tr("Unable to create the folder"));
                 return;
-            waitAction=true;
-            ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to set destination folder time: "+pathList.first().destination.absoluteFilePath());
-            emit errorOnFolder(pathList.first().source,tr("Unable to set time"));
-            return;
+            }
         }
-    if(doRightTransfer && pathList.first().actionType!=ActionType_RealMove)
-    {
-        QFile source(pathList.first().source.absoluteFilePath());
-        QFile destination(pathList.first().destination.absoluteFilePath());
-        if(!destination.setPermissions(source.permissions()))
+        if(doTheDateTransfer)
+            if(!writeFileDateTime(pathList.first().destination))
+            {
+                if(stopIt)
+                    return;
+                waitAction=true;
+                ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to set destination folder time: "+pathList.first().destination.absoluteFilePath());
+                emit errorOnFolder(pathList.first().source,tr("Unable to set time"));
+                return;
+            }
+        if(doRightTransfer && pathList.first().actionType!=ActionType_RealMove)
         {
-            if(stopIt)
+            QFile source(pathList.first().source.absoluteFilePath());
+            QFile destination(pathList.first().destination.absoluteFilePath());
+            if(!destination.setPermissions(source.permissions()))
+            {
+                if(stopIt)
+                    return;
+                waitAction=true;
+                ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to set the right: "+pathList.first().destination.absoluteFilePath());
+                emit errorOnFolder(pathList.first().source,tr("Unable to set the right"));
                 return;
-            waitAction=true;
-            ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to set the right: "+pathList.first().destination.absoluteFilePath());
-            emit errorOnFolder(pathList.first().source,tr("Unable to set the right"));
-            return;
+            }
         }
-    }
-    if(pathList.first().actionType==ActionType_MovePath)
-    {
-        if(!rmpath(pathList.first().source.absoluteFilePath()))
+        if(pathList.first().actionType==ActionType_MovePath)
         {
-            if(stopIt)
+            if(!rmpath(pathList.first().source.absoluteFilePath()))
+            {
+                if(stopIt)
+                    return;
+                waitAction=true;
+                ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to remove the source folder: "+pathList.first().destination.absoluteFilePath());
+                emit errorOnFolder(pathList.first().source,tr("Unable to remove"));
                 return;
-            waitAction=true;
-            ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"Unable to remove the source folder: "+pathList.first().destination.absoluteFilePath());
-            emit errorOnFolder(pathList.first().source,tr("Unable to remove"));
-            return;
+            }
         }
     }
     pathList.removeFirst();
@@ -187,7 +215,7 @@
     this->keepDate=keepDate;
 }
 
-bool MkPath::rmpath(const QDir &dir)
+bool MkPath::rmpath(const QDir &dir,const bool &toSync)
 {
     if(!dir.exists())
         return true;
@@ -198,8 +226,20 @@
         QFileInfo fileInfo(list.at(i));
         if(!fileInfo.isDir())
         {
-            ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"found a file: "+fileInfo.fileName());
-            allHaveWork=false;
+            if(toSync)
+            {
+                QFile file(fileInfo.absoluteFilePath());
+                if(!file.remove())
+                {
+                    ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"unable to remove a file: "+fileInfo.absoluteFilePath()+", due to: "+file.errorString());
+                    allHaveWork=false;
+                }
+            }
+            else
+            {
+                ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Warning,"found a file: "+fileInfo.fileName());
+                allHaveWork=false;
+            }
         }
         else
         {
diff -U 3 -H -d -r -N -- Ultracopier/MkPath.h Rsync/MkPath.h
--- Ultracopier/MkPath.h	2013-03-22 13:28:53.000000000 +0100
+++ Rsync/MkPath.h	2013-03-27 09:30:34.046245103 +0100
@@ -94,7 +94,7 @@
     void internalAddPath(const QFileInfo& source, const QFileInfo& destination,const ActionType &actionType);
     void internalSkip();
     void internalRetry();
-    bool rmpath(const QDir &dir);
+    bool rmpath(const QDir &dir, const bool &toSync=false);
 };
 
 #endif // MKPATH_H
diff -U 3 -H -d -r -N -- Ultracopier/ScanFileOrFolder.cpp Rsync/ScanFileOrFolder.cpp
--- Ultracopier/ScanFileOrFolder.cpp	2013-03-22 13:35:40.000000000 +0100
+++ Rsync/ScanFileOrFolder.cpp	2013-03-28 16:58:39.014607535 +0100
@@ -1,6 +1,7 @@
 #include "ScanFileOrFolder.h"
 #include "TransferThread.h"
 #include <QtGlobal>
+#include <QDateTime>
 
 #ifdef Q_OS_WIN32
 #include <windows.h>
@@ -8,6 +9,7 @@
 
 ScanFileOrFolder::ScanFileOrFolder(Ultracopier::CopyMode mode)
 {
+    rsync               = false;
     moveTheWholeFolder  = true;
     stopped             = true;
     stopIt              = false;
@@ -548,7 +550,17 @@
                     if(!included)
                     {}
                     else
-                        emit fileTransfer(fileInfo,destination.absoluteFilePath()+"/"+fileInfo.fileName(),mode);
+                    {
+                        bool sendToTransfer=false;
+                        if(!rsync)
+                            sendToTransfer=true;
+                        else if(!QFile::exists(destination.absoluteFilePath()+"/"+fileInfo.fileName()))
+                            sendToTransfer=true;
+                        else if(fileInfo.lastModified()!=QFileInfo(destination.absoluteFilePath()+"/"+fileInfo.fileName()).lastModified())
+                            sendToTransfer=true;
+                        if(sendToTransfer)
+                            emit fileTransfer(fileInfo.absoluteFilePath(),destination.absoluteFilePath()+"/"+fileInfo.fileName(),mode);
+                    }
                 }
             }
         }
@@ -558,9 +570,42 @@
                 //listFolder(source,destination,suffixPath+fileInfo.fileName()+QDir::separator());
                 listFolder(fileInfo,destination.absoluteFilePath()+"/"+fileInfo.fileName());//put unix separator because it's transformed into that's under windows too
             else
-                emit fileTransfer(fileInfo,destination.absoluteFilePath()+"/"+fileInfo.fileName(),mode);
+            {
+                bool sendToTransfer=false;
+                if(!rsync)
+                    sendToTransfer=true;
+                else if(!QFile::exists(destination.absoluteFilePath()+"/"+fileInfo.fileName()))
+                    sendToTransfer=true;
+                else if(fileInfo.lastModified()!=QFileInfo(destination.absoluteFilePath()+"/"+fileInfo.fileName()).lastModified())
+                    sendToTransfer=true;
+                if(sendToTransfer)
+                    emit fileTransfer(fileInfo.absoluteFilePath(),destination.absoluteFilePath()+"/"+fileInfo.fileName(),mode);
+            }
         }
     }
+    if(rsync)
+    {
+        //check the reverse path here
+        QFileInfoList entryListDestination=QDir(destination.absoluteFilePath()).entryInfoList(QDir::AllEntries|QDir::NoDotAndDotDot|QDir::Hidden|QDir::System,QDir::DirsFirst|QDir::Name|QDir::IgnoreCase);//possible wait time here
+        int sizeEntryListDestination=entryListDestination.size();
+        int index=0;
+        for (int indexDestination=0;indexDestination<sizeEntryListDestination;++indexDestination)
+        {
+            index=0;
+            while(index<sizeEntryList)
+            {
+                if(entryListDestination.at(indexDestination).fileName()==entryList.at(index).fileName())
+                    break;
+                index++;
+            }
+            if(index==sizeEntryList)
+            {
+                //then not found, need be remove
+                emit addToRmForRsync(entryListDestination.at(indexDestination));
+            }
+         }
+         return;
+    }
     if(mode==Ultracopier::Move)
     {
         ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,"source: "+source.absoluteFilePath()+", sizeEntryList: "+QString::number(sizeEntryList));
@@ -584,3 +629,10 @@
 {
     this->moveTheWholeFolder=moveTheWholeFolder;
 }
+
+/// \brief set rsync
+void ScanFileOrFolder::setRsync(const bool rsync)
+{
+    ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,"set rsync: "+QString::number(rsync));
+    this->rsync=rsync;
+}
diff -U 3 -H -d -r -N -- Ultracopier/ScanFileOrFolder.h Rsync/ScanFileOrFolder.h
--- Ultracopier/ScanFileOrFolder.h	2013-03-22 13:23:18.000000000 +0100
+++ Rsync/ScanFileOrFolder.h	2013-03-27 09:58:42.356175989 +0100
@@ -38,6 +38,7 @@
     void setCheckDestinationFolderExists(const bool checkDestinationFolderExists);
     void setRenamingRules(QString firstRenamingRule,QString otherRenamingRule);
     void setMoveTheWholeFolder(const bool &moveTheWholeFolder);
+    void setRsync(const bool rsync);
 signals:
     void fileTransfer(const QFileInfo &source,const QFileInfo &destination,const Ultracopier::CopyMode &mode);
     /// \brief To debug source
@@ -50,6 +51,7 @@
     void addToMkPath(const QFileInfo& source,const QFileInfo& destination);
     void addToMovePath(const QFileInfo& source,const QFileInfo& destination, const int& inodeToRemove);
     void addToRealMove(const QFileInfo& source,const QFileInfo& destination);
+    void addToRmForRsync(const QFileInfo& destination);
 public slots:
     void addToList(const QStringList& sources,const QString& destination);
     void setFilters(QList<Filters_rules> include,QList<Filters_rules> exclude);
@@ -70,6 +72,7 @@
     volatile bool		checkDestinationExists;
     QString             newName;
     QRegularExpression	folder_isolation;
+    bool                rsync;
     Ultracopier::CopyMode	mode;
     QList<Filters_rules>	include,exclude;
     QList<Filters_rules>	include_send,exclude_send;
diff -U 3 -H -d -r -N -- Ultracopier/StructEnumDefinition_CopyEngine.h Rsync/StructEnumDefinition_CopyEngine.h
--- Ultracopier/StructEnumDefinition_CopyEngine.h	2013-03-25 16:50:03.000000000 +0100
+++ Rsync/StructEnumDefinition_CopyEngine.h	2013-03-27 09:20:20.155270234 +0100
@@ -107,7 +107,8 @@
 {
     ActionType_MkPath=1,
     ActionType_MovePath=2,
-    ActionType_RealMove=3
+    ActionType_RealMove=3,
+    ActionType_RmSync=4
 };
 
 struct Diskspace
diff -U 3 -H -d -r -N -- Ultracopier/TransferThread.cpp Rsync/TransferThread.cpp
--- Ultracopier/TransferThread.cpp	2013-03-26 18:38:59.000000000 +0100
+++ Rsync/TransferThread.cpp	2013-03-27 10:04:53.727160786 +0100
@@ -25,6 +25,7 @@
     writeError                      = false;
     renameTheOriginalDestination    = false;
     needRemove                      = false;
+    rsync                           = false;
     this->mkpathTransfer            = mkpathTransfer;
     readThread.setWriteThread(&writeThread);
     source.setCaching(false);
@@ -389,7 +390,7 @@
                              .arg(readError)
                              .arg(writeError)
                              );
-    if(alwaysDoFileExistsAction==FileExists_Overwrite || readError || writeError)
+    if(alwaysDoFileExistsAction==FileExists_Overwrite || rsync || readError || writeError)
         return false;
     bool destinationExists;
     ULTRACOPIER_DEBUGCONSOLE(Ultracopier::DebugLevel_Notice,"["+QString::number(id)+"] time to first FS access");
@@ -1636,6 +1637,12 @@
     driveManagement.setDrive(mountSysPoint,driveType);
 }
 
+/// \brief set rsync
+void TransferThread::setRsync(const bool rsync)
+{
+    this->rsync=rsync;
+}
+
 void TransferThread::set_osBufferLimit(unsigned int osBufferLimit)
 {
     this->osBufferLimit=osBufferLimit;
diff -U 3 -H -d -r -N -- Ultracopier/TransferThread.h Rsync/TransferThread.h
--- Ultracopier/TransferThread.h	2013-03-23 21:59:47.000000000 +0100
+++ Rsync/TransferThread.h	2013-03-27 09:38:05.451226624 +0100
@@ -132,6 +132,7 @@
     void putAtBottom();
 
     void setDrive(const QStringList &mountSysPoint,const QList<QStorageInfo::DriveType> &driveType);
+    void setRsync(const bool rsync);
 
     void set_osBufferLimit(unsigned int osBufferLimit);
     void setRenamingRules(QString firstRenamingRule,QString otherRenamingRule);
@@ -182,6 +183,7 @@
     QString			destination;*/
     Ultracopier::CopyMode		mode;
     bool			doRightTransfer;
+    bool            rsync;
     bool			keepDate;
     //ready = open + ready to operation (no error to resolv)
     bool			readIsReadyVariable;
diff -U 3 -H -d -r -N -- Ultracopier/copyEngineOptions.ui Rsync/copyEngineOptions.ui
--- Ultracopier/copyEngineOptions.ui	2013-03-26 13:37:09.000000000 +0100
+++ Rsync/copyEngineOptions.ui	2013-03-27 10:05:46.174158639 +0100
@@ -20,7 +20,7 @@
    <item>
     <widget class="QToolBox" name="toolBox">
      <property name="currentIndex">
-      <number>4</number>
+      <number>0</number>
      </property>
      <widget class="QWidget" name="page_trasnfer">
       <property name="geometry">
@@ -62,6 +62,9 @@
          </property>
         </widget>
        </item>
+       <item row="2" column="2">
+        <widget class="QCheckBox" name="autoStart"/>
+       </item>
        <item row="1" column="2">
         <widget class="QCheckBox" name="keepDate"/>
        </item>
@@ -72,10 +75,14 @@
          </property>
         </widget>
        </item>
-       <item row="2" column="2">
-        <widget class="QCheckBox" name="autoStart"/>
+       <item row="4" column="2">
+        <widget class="QCheckBox" name="followTheStrictOrder">
+         <property name="toolTip">
+          <string>Less performance if checked</string>
+         </property>
+        </widget>
        </item>
-       <item row="5" column="1">
+       <item row="6" column="1">
         <spacer name="verticalSpacer_6">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
@@ -98,13 +105,16 @@
          </property>
         </widget>
        </item>
-       <item row="4" column="2">
-        <widget class="QCheckBox" name="followTheStrictOrder">
-         <property name="toolTip">
-          <string>Less performance if checked</string>
+       <item row="5" column="0">
+        <widget class="QLabel" name="label_rsync">
+         <property name="text">
+          <string>Rsync</string>
          </property>
         </widget>
        </item>
+       <item row="5" column="2">
+        <widget class="QCheckBox" name="rsync"/>
+       </item>
       </layout>
      </widget>
      <widget class="QWidget" name="page_error_collision">
@@ -112,8 +122,8 @@
        <rect>
         <x>0</x>
         <y>0</y>
-        <width>791</width>
-        <height>364</height>
+        <width>506</width>
+        <height>218</height>
        </rect>
       </property>
       <attribute name="label">
@@ -316,8 +326,8 @@
        <rect>
         <x>0</x>
         <y>0</y>
-        <width>791</width>
-        <height>364</height>
+        <width>172</width>
+        <height>119</height>
        </rect>
       </property>
       <attribute name="label">
@@ -387,8 +397,8 @@
        <rect>
         <x>0</x>
         <y>0</y>
-        <width>791</width>
-        <height>364</height>
+        <width>332</width>
+        <height>216</height>
        </rect>
       </property>
       <attribute name="label">
@@ -571,8 +581,8 @@
        <rect>
         <x>0</x>
         <y>0</y>
-        <width>791</width>
-        <height>364</height>
+        <width>445</width>
+        <height>89</height>
        </rect>
       </property>
       <attribute name="label">
diff -U 3 -H -d -r -N -- Ultracopier/copyEngineResources.qrc Rsync/copyEngineResources.qrc
--- Ultracopier/copyEngineResources.qrc	2013-03-27 11:57:38.000000000 +0100
+++ Rsync/copyEngineResources.qrc	2013-03-27 12:01:48.468873622 +0100
@@ -1,5 +1,5 @@
 <RCC>
-    <qresource prefix="/CopyEngine/Ultracopier">
+    <qresource prefix="/CopyEngine/Rsync">
         <file>resources/add.png</file>
         <file>resources/edit.png</file>
         <file>resources/remove.png</file>
diff -U 3 -H -d -r -N -- Ultracopier/documentation.dox Rsync/documentation.dox
--- Ultracopier/documentation.dox	2013-01-16 10:30:03.000000000 +0100
+++ Rsync/documentation.dox	2013-03-27 09:13:24.715287241 +0100
@@ -10,7 +10,8 @@
 
   \section mainpage_overview Overview
 
-  The default copy engine for Ultracopier. Support file/folder, copy/move, ...\n
+  The rsync copy engine for Ultracopier. Support file/folder, copy/move, ...\n
+  Thanks to this engine, Ultracopier is transformed into GUI + rsync.
   More informations on <a href="http://ultracopier-wiki.first-world.info/">the wiki of ultracopier</a>.
 
   \section mainpage_platforms Platforms
diff -U 3 -H -d -r -N -- Ultracopier/informations.xml Rsync/informations.xml
--- Ultracopier/informations.xml	2013-02-28 12:56:19.000000000 +0100
+++ Rsync/informations.xml	2013-03-27 10:03:15.911164790 +0100
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <package>
-	<title xml:lang="en"><![CDATA[Copy engine of Ultracopier]]></title><!-- english is required -->
-	<title xml:lang="fr"><![CDATA[Moteur de copie d'ultracopier]]></title>
+	<title xml:lang="en"><![CDATA[Rsync copy engine]]></title><!-- english is required -->
+	<title xml:lang="fr"><![CDATA[Moteur de copie rsync]]></title>
 	<!-- What kind of plugin this is -->
 	<category>CopyEngine</category>
 	<!-- Who wrote this plugin -->
@@ -14,12 +14,12 @@
 	<!-- the architecture code of this plugin, found PlatformMacro.h into ultracopier source -->
 	<architecture>linux-x86_64-pc</architecture>
 	<!-- Detailed description -->
-	<description xml:lang="en"><![CDATA[Copy engine of Ultracopier, do in Qt for all platform, and copy by stream]]></description>
-	<description xml:lang="fr"><![CDATA[Moteur de copie d'ultracopier, fait en Qt pour toutes les plateformes, et copie par stream]]></description>
+	<description xml:lang="en"><![CDATA[Rsync copy engine, variant of the official copy engine to provide simple way to do backups]]></description>
+	<description xml:lang="fr"><![CDATA[Moteur de copie rsync, variante du moteur de copie officiel pour fournir une maniére simple de faire des backups]]></description>
 	<!-- Version of this release of this plugin, need be like that's: A.B.C.D, where A, B, C and D is number -->
 	<version>1.0.0.0</version>
 	<!-- This internal name should never change, because it is used to detect when a particular plugin is updated. It must comprise only lower case ASCII characters (a-z), numerical digits (0-9), "-", "." or "_", and it must be be unique within the category. And have size lower than 64 char. -->
-	<name>Ultracopier</name>
+	<name>Rsync</name>
 	<!-- Dependency checking. This is used to check when a plugin may not be compatible with an updated version of either Ultracopier or another plugin. This example only checks Ultracopier. -->
 	<dependencies><![CDATA[
 	=ultracopier-1.0
